core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1103,9 +1103,14 @@ class DAGScheduler(
         // multiple tasks running concurrently on different executors). In that case, it is possible
         // the fetch failure has already been handled by the scheduler.
         if (runningStages.contains(failedStage)) {
-          logInfo(s"Marking $failedStage (${failedStage.name}) as failed " +
-            s"due to a fetch failure from $mapStage (${mapStage.name})")
-          markStageAsFinished(failedStage, Some(failureMessage))
+          if (failedStage.attemptId - 1 > task.stageAttemptId) {
```

Comment:
I'm confused about how this works.  Doesn't the stage still get added to failed stages on line 1149, so it will still be resubmitted?